merge/update.go

```diff
@@ -62,19 +62,13 @@ func (s *Updater) update(oldObject, newObject *typed.TypedValue, version fieldpa
 			var err error
 			versionedOldObject, err := s.Converter.Convert(oldObject, managerSet.APIVersion())
```

Comment:
I also expected to see the check that the API version was non-empty before trying to call conversion at all

Reply:
Should we make this validation somewhere else? That really shouldn't happen.

Reply:
> That really shouldn't happen.

Isn't that exactly what happened in https://github.com/kubernetes/kubernetes/issues/91006#issuecomment-628293331?

Reply:
:-)

It happened because we used to push such an object for "clearing managedFields". Since we're moving away from that model, it shouldn't happen anymore!

Reply:
if it was an approach we documented and people have picked up using, I'd expect us to handle it

Reply:
https://kubernetes.io/docs/reference/using-api/api-concepts/#clearing-managedfields

Even if we want to keep that pattern, we probably shouldn't reach that low in the stack with the empty `apiVersion`.

Reply:
I thought we decided to check for that when decoding managedFields in the fieldManager.